src/ResourceManager/IotCentral/Commands.IotCentral/Common/IotCentralBaseCmdlet.cs

```diff
@@ -0,0 +1,84 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources;
+using Microsoft.Azure.Management.IotCentral;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.IotCentral.Common
+{
+    public abstract class IotCentralBaseCmdlet : AzureRMCmdlet
+    {
+        private IIotCentralClient iotCentralClient;
+
+        private IResourceManagementClient resourceManagementClient;
+
+        protected const string InteractiveIotCentralParameterSet = "InteractiveIotCentralParameterSet";
+        protected const string ResourceIdParameterSet = "ResourceIdParameterSet";
+        protected const string InputObjectParameterSet = "InputObjectParameterSet";
+
+        [Parameter(
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
```

Comment:
@cle5eland I would remove the `ValueFromPipelineByPropertyName` property from the `-ResourceGroupName` and `-Name` parameters to ensure that users are piping the result of `Get-AzureRmResource` to the parameter set containing `-ResourceId` rather than the set containing `-ResourceGroupName` and `-Name`

Reply:
Good call, fixed.